src/Features/Core/Portable/AddImport/Remote/AbstractAddImportFeatureService_Remote.cs

```diff
@@ -22,7 +22,7 @@ internal abstract partial class AbstractAddImportFeatureService<TSimpleNameSynta
         /// back to VS, which will then bounce back out to OOP to perform the Nuget/ReferenceAssembly
         /// portion of the search.  Ideally we could keep this all OOP.
         /// </summary>
-        private class RemoteSymbolSearchService : IRemoteSymbolSearchUpdateEngine
+        internal sealed class RemoteSymbolSearchService : IRemoteSymbolSearchUpdateEngine
```

Comment:
how come?

Reply:
You mean why internal? Because this is an type whose methods we invoke remotely and I use references to its methods in `nameof()` in the remote invocation call-sites. Previously there was no enforcement that the call-site name is actually then name of the method on the target object.

Reply:
Ah.  so there is *some* enforcement.  Specifically, the *caller* is supposed to do `nameof(IRemoteHoosyWhatsit.DoTheThingAsync)`.  Then `RemoteHoosyWhatsit` implements `IRemoteHoosyWhatsit.DoTheThingAsync`.  so you get the tight binding throught he interface.

If you're *not* goin to use this pattern, then you can literally just get rid of hte interfaces.  They were invented *purely* to let us tie these two things together (i.e. ensuring the caller knows the name, and that the method exists).

Reply:
*That said*, i believe @sharwell was also saying that the RPC supports dynamic proxies for interfaces.  so we could literally just use interfaces here and call into those, and all this goop goes away.